driver/src/test/functional/com/mongodb/reactivestreams/client/gridfs/GridFSPublisherSpecification.groovy

```diff
@@ -378,43 +408,58 @@ class GridFSPublisherSpecification extends FunctionalSpecification {
     }
 
     def 'should cleanup when unsubscribing'() {
-        when:
-        def contentBytes = multiChunkString as byte[]
+        given:
+        def contentSize = 1024 * 1024
+        def contentBytes = new byte[contentSize]
+        new SecureRandom().nextBytes(contentBytes)
+        def options = new GridFSUploadOptions().chunkSizeBytes(1024)
+        def data = (0..1024).collect { ByteBuffer.wrap(contentBytes) }
+        def subscriber = new Subscriber<ObjectId>() {
+            Subscription subscription
+            @Override
+            void onSubscribe(final Subscription s) {
+                subscription = s
+            }
 
-        then:
-        run(filesCollection.&countDocuments) == 0
+            @Override
+            void onNext(final ObjectId o) {
+            }
+
+            @Override
+            void onError(final Throwable t) {
+            }
+
+            @Override
+            void onComplete() {
+            }
+        }
 
         when:
-        def subscriber = new ObservableSubscriber()
-        gridFSBucket.uploadFromPublisher('myFile', createPublisher(ByteBuffer.wrap(contentBytes), ByteBuffer.wrap(contentBytes)))
+        gridFSBucket.uploadFromPublisher('myFile', createPublisher(*data), options)
```

Comment:
What does `*data` mean in this context?

Reply:
Its a way to explode a list to varargs.

`def data = (0..1024).collect { ByteBuffer.wrap(contentBytes) }`

Data is a list that contains 1024 ByteBuffers - so here we create a publisher with 1024 elements.